articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+
+## Create a new Decorator
+
+Typically you would create a new Decorator in a new Intent Architect Application which will reference a Template from another Module or Intent Architect Application. As such we will be doing that.
+
+![Startup](images/create-new-decorator/MyDecoratorModuleStartup.png)
+
+Go ahead and create your Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
+
+Once the generation is done, create a new Decorator called `StaticFileServerDecorator`.
+
+![New Decroator](images/create-new-decorator/NewDecorator.png)
+
+We need to specify the Contract name for that Decorator so that it will know for which Template it is meant to Decorate.
+
+To recap, the Template we're trying to Decorate is the `StartupTemplate` and it requires a Decorator that adheres to this interface:
+
+```csharp
+public interface IStartupTemplateContract : ITemplateDecorator
+{
+    string ConfigureCode();
+}
+```
+
+It has a Full Namespace of: `MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract`
+
+So that is the value that we need to specify in the `Full TypeName` field for that Decorator.
+
+Also, click on the `Declare Usings` flag in the `Decorator Settings` panel.
+
+![Add Type FullName](images/create-new-decorator/AddFQDN.png)
+
+Run the code generation.
+
+![Code Generation Staging](images/create-new-decorator/CodeGenerationStaging.png)
+
+Open Visual Studio and navigate to where the code has been generated and open up the `MyCompany.MyDecoratorModule` solution.
+
+You will notice that it has generated a Decorator as well as a Registration for the Decorator.
+
+![Solution Tree Structure](images/create-new-decorator/SolutionTreeStructure.png)
+
+Although, you will notice that the IDE reports a problem with our namespace we gave it.
+
+![IDE Error](images/create-new-decorator/IDEError.png)
+
+## Ensure the "blueprint" (or contract) assembly is referenced
+
+To fix this issue, we need to make sure that we reference the assembly that contains `IStartupTemplateContract`.
+
+In our case this is simple. We add a project reference to that solution.
+
+![Add Existing Project](images/create-new-decorator/AddExistingProject.png)
+
+Let's add that module as an existing project into our current solution.
+
+![Add Project Reference](images/create-new-decorator/AddProjectReference.png)
+
+Now we add that project as a reference to our new module project.
+
+## Implement the Decorator
+
+This should resolve our initial problem which is that the interface was not found but now we need to implement that interface, so we're not out of the woods yet.
+
+![Go To Implement Interface Command](images/create-new-decorator/GoToImplementInterfaceCommand.png)
+
+Once we instructed Visual Studio to implement this interface, you will notice that it added two things:
+
+```csharp
```

Comment:
Towards keeping the how-to as concise as possible, let's make the code work off the bat.

Reply:
What do you mean?

Reply:
My thinking is that the code example here should already be done correctly fully. We don't necessarily need to show what it looked like by default after the code-gen, they would see it themselves, it's more concise to just have written below on what it needs to be.

(In case you're unfamiliar with the idiom "off-the-bat" which I might have also used slightly wrong, you can see an explanation [here](https://idioms.thefreedictionary.com/off+the+bat)).

Reply:
I was unsure what you meant based on the context. So the idea, like mentioned in a previous comment above (https://github.com/IntentSoftware/Docs/pull/1#discussion_r365611409) is to assume that developers are going to run into issues, so I wanted to make it as smooth as possible for them to get to the end destination by adding a little more content. Until Intent can more smoothly do more or we know that most developers know how to work with C# and Visual Studio, I would keep that there just in case.

Reply:
@dandrejvv, while having dinner, it occurred to me that we need to decide on what assumptions this article makes on the background of the reader. Once we've decided on that, then it becomes quite clear what can be omitted.

From that blog post in regards to how-tos:
> **Practical usability is more valuable than completeness.** Tutorials need to be complete, end-to-end guides; how-to guides do not. They can start and end where it seems appropriate to you. They don’t need to mention everything that there is to mention either, just because it is related to the topic. A bloated how-to guide doesn’t help the user get speedily to their solution.

So, with that in mind, here are my thoughts on assumptions this article can make:
- Decorators are an advanced concept and are probably not going to be taken on by people brand new to Intent.
- We assume the reader already knows about writing modules and templates and feel somewhat comfortable doing so already.
- We can assume basic proficiency in Visual Studio (or their IDE of choice) and C#, so they know:
  - ...how to implement an interface.
  - ...how to reference assemblies or add NuGet packages to projects.
  - ...enough about `usings` and namespaces.

In practical terms, let's imagine someone like any of your colleagues at your current job is already practised at creating and updating their own templates in Intent and now found the need to make a decorator, either for their one of their own template, or someone else's.

If we were standing over their shoulder and telling them what to click and write, what would be the minimum instructions that we would need to give them? And that's what I think this how-to should be.

Also, I think that how-tos like this are consumed by quite senior devs, people like ourselves. Speaking for myself, I generally just want to see as quickly as possibly with as little fluff as possible what needs to be done. For things that are unclear to me, then to save me a Google search, it's very useful within the how-to to have links to reference articles which explain them in depth.

In summary for how-to articles, assume the reader kind of knows a lot, but just wants to see how-to do something.

Reply:
Hi @JonathanLydall , I've sent you a personal response to the message above.